flake.nix

```diff
@@ -0,0 +1,22 @@
+{
+    
+    description = "Kubernetes object analysis with recommendations for improved reliability and security";
+
+    inputs.flake-utils.url = "github:numtide/flake-utils";
+
+    outputs = { self, nixpkgs, flake-utils }:
+        flake-utils.lib.eachDefaultSystem (system:
+            let pkgs = import nixpkgs { inherit system; }; in
+            rec {
+                packages.kube-score = pkgs.buildGoModule {
+                    name = "kube-score";
+                    src = self;
+                    vendorSha256 = "sha256-E9pcJsnoF/SKRCjrHZY8Ybd8kV1F3FYwdnLJ0mHyRLA="; # master
```

Comment:
As kube-score is using automated tooling for dependency management, especially for upgrades, I’d prefer if we didn’t have to specify this checksum. Is there a way to avoid it?

If it’s there, I’d assume that it will be out of date and that builds will start to fail as soon as `go.mod` is updated, or am I wrong here?

Reply:
Your assumptions are both right. `vendorSha256 = null` does disable the paranoic check. _However_, paranoic checks are considered a good thing looking at it on the aggregate level. _And_, when I tried building it without it, there seems to be a bug somwhere in the enviornment when resolving some kubeapiserver packages to `.`. I can have another go to this, but it looked rather obscure. Not sure if this is abug on nix tooling's side. Maybe not.

Reply:
Unfortunately it looks like nixpkgs.buildGoPackage is built in a way so that it only can "trust" dependencies if they are vendored and checked into source.
See:
https://github.com/NixOS/nixpkgs/blob/0a2cff8e74baacfa8da9bad843699fe71a8c3053/pkgs/development/go-modules/generic/default.nix#L18-L20

So I'm afraid it looks like this is not possible, since I assume you neither want to vendor sources.

Reply:
I think the root cause its that nix doesn't want to offload source of truth to another tooling. Evan though go arguably does a good job in reproducability and validation with go.sum. It's just that since those resources are fetched over the network, nix want's to make sure (on it's own) that it's in the box what it pretends to be.

Reply:
Maybe it's possible to do something similar to what we do with Rust, namely get the dependencies from `Cargo.lock`. That way you don't need a `vendorSha256` in your flake.nix that could get out of date. See https://github.com/edolstra/import-cargo.

Reply:
That sounds like it would be an excellent leverage for the nix community and would naturally lowers barrier of adoption.